### PR TITLE
use monotonic clock for duration

### PR DESCRIPTION
**Pull request by dirk-thomas**
Same as colcon/colcon-core#201.

### CODE REVIEW

File: colcon_notification/event_handler/desktop_notification/__init__.py [hunk 1]
```diff
@@ -33,7 +33,6 @@ def __init__(self):  # noqa: D107
         super().__init__()
         satisfies_version(
             EventHandlerExtensionPoint.EXTENSION_POINT_VERSION, '^1.0')
-        self._start_time = time.time()
```
Comment (Karsten1987) [inline colcon_notification/event_handler/desktop_notification/__init__.py]: is that change on purpose? Or should this also be changed to `monotonic()`?

File: colcon_notification/event_handler/status.py [hunk 2]
```diff
@@ -114,12 +114,12 @@ def __call__(self, event):  # noqa: D102
         elif isinstance(data, JobEnded):
             job = event[1]
             self._ended[job] = self._running[job]
-            self._ended[job]['end_time'] = time.time()
+            self._ended[job]['end_time'] = time.monotonic()
```
Comment (clalancette) [inline colcon_notification/event_handler/status.py]: It looks like the `end_time` key isn't used; consider removing it.  On the other hand, it has nice symmetry with the `start_time` key in the `_running` dictionary, so I don't feel strongly about it.

File: colcon_notification/event_handler/desktop_notification/__init__.py (hunk 1 above)
Comment (dirk-thomas) [reply colcon_notification/event_handler/desktop_notification/__init__.py]: I didn't see where this field was used anywhere in this class and there are no subclasses of it either. Hence I removed it.
